content/en/blog/2021/proxyless-grpc/index.md

```diff
@@ -0,0 +1,402 @@
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: echo-mtls
+  namespace: echo-grpc
+spec:
+  host: echo.echo-grpc.svc.cluster.local
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+EOF
+{{< /text >}}
+
+Now an attempt to call the server that is not yet configured for mTLS will fail.
+
+{{< text bash >}}
+$ grpcurl -plaintext -d '{"url": "xds:///echo.echo-grpc.svc.cluster.local:7070"}' :17171 proto.EchoTestService/ForwardEcho | jq -r '.output | join("")'
+Handling connection for 17171
+ERROR:
+Code: Unknown
+Message: 1/1 requests had errors; first error: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
+{{< /text >}}
+
+To enable server-size mTLS, apply a `PeerAuthentication`.
+
+{{< warning >}}
+The following policy forces STRICT mTLS for the entire namespace.
+{{< /warning >}}
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -f -
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: echo-mtls
+  namespace: echo-grpc
+spec:
+  mtls:
+    mode: STRICT
+EOF
+{{< /text >}}
+
+Requests will start to succeed after applying the policy.
+
+{{< text bash >}}
+$ grpcurl -plaintext -d '{"url": "xds:///echo.echo-grpc.svc.cluster.local:7070"}' :17171 proto.EchoTestService/ForwardEcho | jq -r '.output | join("")'
+Handling connection for 17171
+[0] grpcecho.Echo(&{xds:///echo.echo-grpc.svc.cluster.local:7070 map[] 0  5s false })
+[0 body] x-request-id=0
+[0 body] Host=echo.echo-grpc.svc.cluster.local:7070
+[0 body] content-type=application/grpc
+[0 body] user-agent=grpc-go/1.39.1
+[0 body] StatusCode=200
+[0 body] ServiceVersion=v1
+[0 body] ServicePort=17070
+[0 body] Cluster=
+[0 body] IP=10.68.1.18
+[0 body] IstioVersion=
+[0 body] Echo=
+[0 body] Hostname=echo-v1-7cf5b76586-z5p8l
+{{< /text >}}
+
+## Limitations
+
+The initial release comes with several limitations that may be fixed in a future version:
+
+* Auto-mTLS isn't supported, and permissive mode isn't supported. Instead we require explicit mTLS configuration with
+  `STRICT` on the server and `ISTIO_MUTUAL` on the client. Envoy can be used during the migration to `STRICT`.
+* `grpc.Serve(listener)` or `grpc.Dial("xds:///...")` called before the bootstrap is written or xDS proxy is ready can
+  cause a failure. `holdApplicationUntilProxyStarts` can be used to work around this, or the application can be more
+  robust to these failures.
+* If the xDS-enabled gRPC server uses mTLS then you will need to make sure your health checks can work around this.
+  Either a separate port should be used, or your health-checking client needs a way to get the proper client
+  certificates.
+* The implementation of xDS in gRPC does not match Envoys. Certain behaviors may be different, and some features may
+  be missing. The [`gRFCs`](https://github.com/grpc/proposal) may provide more detail. Make sure to test that any Istio
```

Comment:
Drive by comment: https://github.com/grpc/grpc/blob/master/doc/grpc_xds_features.md may be a bit more useful (in addition to/instead of the gRFC link)

Reply:
Oh, I see that is linked to above. Seemed it would sorta be useful here, but I question whether it really needs to be linked to again.  I guess nevermind.

Reply:
```suggestion
  be missing. The  feature status for gRPC is listed [here](https://github.com/grpc/grpc/blob/master/doc/grpc_xds_features.md ). Make sure to test that any Istio
```

Reply:
How is this?

Reply:
That would be fine. I'd probably word it more like before to avoid "here":

> The \[feature status for gRPC](link) provides more detail.

But that's editorial.

Reply:
```suggestion
  be missing. The [feature status for gRPC](https://github.com/grpc/grpc/blob/master/doc/grpc_xds_features.md) provides more detail. Make sure to test that any Istio
```